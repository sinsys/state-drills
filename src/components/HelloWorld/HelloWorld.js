import React from 'react';
import './HelloWorld.css';

class HelloWorld extends React.Component {
	static defaultProps = {
		who: "World",
		options: ["World", "friend", "React", "test", "stuff", "blah"]
	}
	constructor(props){
		super(props);
		this.state = {
			who: this.props.who
		}
	}
	updateText = (option) => {
		this.setState ({
			who: option
		})
	}
	render() {
		return (
			<div className="HelloWorld-wrapper">
				<h2>Hello World + Interactivity</h2>
				<p>Hello, {this.state.who}!</p>
				{this.props.options.map((option) => (
					<button 
						key={option}
						onClick={() => this.updateText(option)}>
						{option}
					</button>
				))}
			</div>
		)
	}
}

export default HelloWorld;