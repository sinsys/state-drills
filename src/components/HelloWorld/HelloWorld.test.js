import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import HelloWorld from './HelloWorld';

describe('HelloWorld Component', () => {
	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<HelloWorld />, div);
		ReactDOM.unmountComponentAtNode(div);
	})

	it('ui renders as expected', () => {
		expect(
			renderer.create(<HelloWorld />).toJSON()
		).toMatchSnapshot()
	});
});