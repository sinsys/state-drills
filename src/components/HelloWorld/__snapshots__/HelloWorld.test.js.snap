// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld Component ui renders as expected 1`] = `
<div
  className="HelloWorld-wrapper"
>
  <h2>
    Hello World + Interactivity
  </h2>
  <p>
    Hello, 
    World
    !
  </p>
  <button
    onClick={[Function]}
  >
    World
  </button>
  <button
    onClick={[Function]}
  >
    friend
  </button>
  <button
    onClick={[Function]}
  >
    React
  </button>
  <button
    onClick={[Function]}
  >
    test
  </button>
  <button
    onClick={[Function]}
  >
    stuff
  </button>
  <button
    onClick={[Function]}
  >
    blah
  </button>
</div>
`;
